### PR TITLE
Fix broken challenges

### PR DESCRIPTION
**Pull request by crookedneighbor**
A migration file for retrieving broken challenge tasks and _trying_ to correct them.

### CODE REVIEW

File: migrations/20160524_fix_challenges.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+function _logger (type, color) {
+  return function () {
+    let args = Array.from(arguments).map(arg => chalk[color](arg));
+    console[type].apply(null, args);
+  }
+}
+
+// PROD: Enable prod db
+// const NODE_DB_URI = 'mongodb://username:password@dsXXXXXX-a0.mlab.com:XXXXX,dsXXXXXX-a1.mlab.com:XXXXX/habitica?replicaSet=rs-dsXXXXXX';
+const NODE_DB_URI = 'mongodb://localhost/new-prod-copy';
```
Comment (crookedneighbor) [inline migrations/20160524_fix_challenges.js]: Change this to a local backup of the prod db for testing.

File: migrations/20160524_fix_challenges.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+// Cached ids from running the findBrokenChallengeTasks query on a local copy of the db
+const TASK_IDS = require('../taskids.json');
```
Comment (crookedneighbor) [inline migrations/20160524_fix_challenges.js]: Put the json file in the root of the repo or change this path to it.

File: migrations/20160524_fix_challenges.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+function unique (array) {
+  return Array.from(new Set(array));
+}
+
+function findBrokenChallengeTasks () {
+  logger.info('Looking for broken tasks...');
+
+  // return db.collection('tasks').find({'challenge.broken': 'CHALLENGE_TASK_NOT_FOUND'}).toArray()
```
Comment (crookedneighbor) [inline migrations/20160524_fix_challenges.js]: If you don't want to use the pre-cached ids, uncomment this and comment the line below.

**Comment by crookedneighbor** [comment]
Fixes #7420

**Comment by crookedneighbor** [comment]
Meged in with https://github.com/HabitRPG/habitrpg/commit/038dd64b12f11f6af3bcd719d69c99828986b9ba

**Comment by lucas-chu** [comment]
Still have this problem
image

**Comment by SabreCat** [comment]
@lucas-chu Commenting on an old pull request isn't really the best way to communicate with the team--I'd suggest using the Report a Bug link on the website under the Help menu!

**Pull request closed without merging**